pkg/cvo/cvo_scenarios_test.go

```diff
@@ -620,6 +621,266 @@ func TestCVO_StartupAndSyncUnverifiedPayload(t *testing.T) {
 	expectGet(t, actions[0], "clusterversions", "", "version")
 }
 
+func TestCVO_StartupAndSyncPreconditionFailing(t *testing.T) {
+	o, cvs, client, _, shutdownFn := setupCVOTest("testdata/payloadtest")
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	defer shutdownFn()
+
+	worker := o.configSync.(*SyncWorker)
+	worker.preconditions = []precondition.Precondition{&testPrecondition{SuccessAfter: 100}}
```

Comment:
It wasn't clear reading this test which step checks this, could you make that clearer?

Reply:
added a comment, what do think now? any clearer?